packages/flutter/test/rendering/sliver_fixed_extent_layout_test.dart

```diff
@@ -41,6 +41,42 @@ void main() {
     expect(children[1].attached, false);
     expect(children[2].attached, true);
   });
+
+  group('getMaxChildIndexForScrollOffset', () {
+    const double itemExtent = 600.0;
+    testValidationRounding(1234, 0, 0, 'When item extent is 0');
```

Comment:
These tests are a bit difficult to understand, can you refactor them into separate tests that use a common method? Or one large test that breaks down each case?

Reply:
I create a test per case. Thanks